src/components/organisms/PageHeader/ClusterSelection.styled.tsx

```diff
@@ -1,42 +1,101 @@
-import {Button as RawButton, Divider as RawDivider} from 'antd';
+import {ButtonProps, Button as RawButton, Divider as RawDivider} from 'antd';
 
 import {ClusterOutlined as RawClusterOutlined, DownOutlined as RawDownOutlined} from '@ant-design/icons';
 import {
   CheckCircleOutlined as RawCheckCircleOutlined,
   ExclamationCircleOutlined as RawExclamationCircleOutlined,
 } from '@ant-design/icons/lib/icons';
 
+import {isBoolean} from 'lodash';
+import {rgba} from 'polished';
 import styled from 'styled-components';
 
+import {PreviewType} from '@models/appstate';
+
 import Colors from '@styles/Colors';
 
-export const Button = styled(RawButton)`
-  padding: 0;
-  margin: 0;
-  color: ${Colors.blue6};
+interface RawButtonProps extends ButtonProps {
+  isInPreviewMode?: boolean;
+  previewType?: PreviewType;
+}
+interface IAntdIconProps extends React.RefAttributes<HTMLSpanElement> {
+  isInPreviewMode?: boolean;
+  previewType?: PreviewType;
+  isKubeConfigPathValid?: boolean;
+}
+
+export const getPreviewTheme = (
+  fallBackColor: string,
+  previewType?: PreviewType,
+  rgbaRatio?: number,
+  isInPreviewMode?: boolean
+) => {
+  let color = fallBackColor;
+
+  if (previewType === 'cluster') {
+    color = rgbaRatio ? rgba(Colors.volcano, rgbaRatio) : Colors.volcano;
+  }
+  if (previewType === 'helm') {
+    color = rgbaRatio ? rgba(Colors.cyan, rgbaRatio) : Colors.cyan;
+  }
+  if (previewType === 'helm-preview-config') {
+    color = rgbaRatio ? rgba(Colors.cyan, rgbaRatio) : Colors.cyan;
+  }
+  if (previewType === 'kustomization') {
+    color = rgbaRatio ? rgba(Colors.cyan, rgbaRatio) : Colors.cyan;
+  }
 
-  &:hover {
-    color: ${Colors.blue6};
+  return isBoolean(isInPreviewMode) ? (isInPreviewMode ? color : fallBackColor) : color;
+};
+
+export const Button = styled(({children, previewType, isInPreviewMode, ...rest}: RawButtonProps) => (
```

Comment:
Unused properties should be removed?

Reply:
they cause an error if I don't extract them. I couldn't find a way. Do you have an idea @WitoDelnat ?